src/node_report_module.cc

```diff
@@ -0,0 +1,295 @@
+  Local<Value> verbose_unchecked;
+  if (!obj->Get(context, verbosekey).ToLocal(&verbose_unchecked)) return;
+  Local<Boolean> verbose;
+  if (verbose_unchecked->IsUndefined() || verbose_unchecked->IsNull())
+    verbose_unchecked = Boolean::New(env->isolate(), "verbose");
+  verbose = verbose_unchecked.As<Boolean>();
+
+  bool verb = verbose->BooleanValue(context).FromJust();
+
+  if (sync) {
+    static const std::string e = "exception";
+    static const std::string s = "signal";
+    static const std::string f = "fatalerror";
+    for (uint32_t i = 0; i < events->Length(); i++) {
+      Local<Value> v;
+      if (!events->Get(context, i).ToLocal(&v)) return;
+      Local<String> elem;
+      if (!v->ToString(context).ToLocal(&elem)) return;
+      String::Utf8Value buf(env->isolate(), elem);
+      if (*buf == e) {
+        options->report_uncaught_exception = true;
+      } else if (*buf == s) {
+        options->report_on_signal = true;
+      } else if (*buf == f) {
+        options->report_on_fatalerror = true;
+      }
+    }
+    if (*signalstr != nullptr) options->report_signal = *signalstr;
+    if (*filestr != nullptr) options->report_filename = *filestr;
+    if (*pathstr != nullptr) options->report_directory = *pathstr;
+    options->report_verbose = verb;
+  } else {
+    int i = 0;
+    if (options->report_uncaught_exception)
+      if (events
+              ->Set(context,
+                    i++,
+                    FIXED_ONE_BYTE_STRING(env->isolate(), "exception"))
+              .IsNothing())
+        return;
+    if (options->report_on_signal)
+      if (events
+              ->Set(
+                  context, i++, FIXED_ONE_BYTE_STRING(env->isolate(), "signal"))
+              .IsNothing())
+        return;
+    if (options->report_on_fatalerror)
+      if (events
+              ->Set(context,
+                    i,
+                    FIXED_ONE_BYTE_STRING(env->isolate(), "fatalerror"))
+              .IsNothing())
+        return;
+
+    Local<Value> signal_value;
+    Local<Value> file_value;
+    Local<Value> path_value;
+    if (!node::ToV8Value(context, options->report_signal)
+             .ToLocal(&signal_value))
+      return;
+    if (!obj->Set(context, signalkey, signal_value).FromJust()) return;
+
+    if (!node::ToV8Value(context, options->report_filename)
+             .ToLocal(&file_value))
+      return;
+    if (!obj->Set(context, filekey, file_value).FromJust()) return;
+
+    if (!node::ToV8Value(context, options->report_directory)
+             .ToLocal(&path_value))
+      return;
+    if (!obj->Set(context, pathkey, path_value).FromJust()) return;
+
+    if (!obj->Set(context,
+                  verbosekey,
+                  Boolean::New(env->isolate(), options->report_verbose))
+             .FromJust())
+      return;
+  }
+}
+
+static void Initialize(Local<Object> exports,
+                         Local<Value> unused,
+                         Local<Context> context) {
```

Comment:
I think arguments should be aligned. (https://github.com/nodejs/node/blob/master/CPP_STYLE_GUIDE.md#align-function-arguments-vertically)

Reply:
after @addaleax 's suggestions, I am now bluntly submitting the alignments to `make format-cpp` ! Do you strongly feel against it?

Reply:
Nope, I'm good with either.